src/components/TableOfContents/index.tsx

```diff
@@ -44,15 +44,11 @@ const TableOfContents: React.FC<IProps> = ({
 
   if (!isMobile) {
     const getTitleIds = (items: Array<ToCItem>, depth: number): void => {
-      if (depth > (maxDepth ? maxDepth : 1)) return
+      if (depth > (maxDepth ?? 1)) return
```

Comment:
we could add a variable name explaining what this conditional means and use that inside the `if` instead, as its hard to follow what's happening

`x = depth > (maxDepth ?? 1)`, where `x` should be a declarative name explaining the condition

Reply:
Same as above

Reply:
removed fallback (redundant) and added a comment